modules/core/src/main/java/org/apache/ignite/startup/cmdline/CommandLineStartup.java

```diff
@@ -62,10 +82,50 @@
     /** Quite log flag. */
     private static final boolean QUITE;
 
+    /** Command to print Ignite system properties info. */
+    static final String PRINT_PROPS_COMMAND = "-systemProps";
+
+    /** Classes with Ignite system properties. */
+    static final List<Class<?>> PROPS_CLS = new ArrayList<>(Arrays.asList(
+        IgniteSystemProperties.class,
+        ExchangeContext.class,
+        GridCacheMapEntry.class,
+        LocalDeploymentSpi.class,
+        GridCacheDatabaseSharedManager.class,
+
+        // TODO: GG-34386
+        //GridJmxPortFinder.class,
+        PartitionsEvictManager.class,
+        PagesList.class,
+        PagesList.PagesCache.class,
+        GridCacheOffheapManager.class,
+        CacheContinuousQueryEventBuffer.class,
+        CacheContinuousQueryHandler.class,
+        OffheapReadWriteLock.class,
+        TcpCommunicationConfiguration.class
+    ));
+
     /**
      * Static initializer.
      */
     static {
+        String h2TreeCls = "org.apache.ignite.internal.processors.query.h2.database.H2Tree";
```

Comment:
It would be nice to create a new issue that should improve this behavior and address `GridSystemProperties` too

Reply:
Agree with @sk0x50 here: all sources of properties information have to be hardcoded which turns plugins support into a mess.

We need to create a ticket to support information providers of some kind.

Reply:
Created: https://issues.apache.org/jira/browse/IGNITE-16179